From: Jason Dillaman <dillaman@redhat.com>
Date: Mon, 19 Sep 2016 11:24:17 -0400
Subject: journal: send update notification when tag allocated

Signed-off-by: Jason Dillaman <dillaman@redhat.com>
(cherry picked from commit f1cd613e3d8ccb6d05b9adfe1956927991a4f4fe)

Resolves: rhbz#1365648
---
 src/journal/JournalMetadata.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/journal/JournalMetadata.cc b/src/journal/JournalMetadata.cc
index 36dffdf..8688a50 100644
--- a/src/journal/JournalMetadata.cc
+++ b/src/journal/JournalMetadata.cc
@@ -561,6 +561,7 @@ void JournalMetadata::unregister_client(Context *on_finish) {
 
 void JournalMetadata::allocate_tag(uint64_t tag_class, const bufferlist &data,
                                    Tag *tag, Context *on_finish) {
+  on_finish = new C_NotifyUpdate(this, on_finish);
   C_AllocateTag *ctx = new C_AllocateTag(m_cct, m_ioctx, m_oid,
                                          m_async_op_tracker, tag_class,
                                          data, tag, on_finish);
